elastic/kibana
# [Synthetics] Update API key to create monitors

## Summary

Added UI to copy API for project monitors

Scoped different kind of API keys to different permissions.

Update API key privileges to create monitors as well

Added a simple UI in uptime to generate and copy API key which can be used with synthetics agent push command

For users with API Key Permissions
<img width="796" alt="Screen Shot 2022-05-24 at 3 17 34 PM" src="https://user-images.githubusercontent.com/11356435/170118508-904fe7cd-37ba-468c-94a1-d229c148d3bc.png">
<img width="801" alt="Screen Shot 2022-05-24 at 3 15 20 PM" src="https://user-images.githubusercontent.com/11356435/170118513-b6ad9a00-35c0-42d5-840e-a5836afc59d5.png">

For users without API Key Permissions

<img width="1423" alt="Screen Shot 2022-05-24 at 2 58 54 PM" src="https://user-images.githubusercontent.com/11356435/170118676-952371bd-1565-42c6-93a5-5196b366d6ff.png">


### Testing
Testing that we can use this API key to create monitors
--

1. Go to monitor management
2. Create a monitor and copy the curl request from chrome network tab, which was used to create monitor
3. Use tool like postman to import the curl request
4. Update Authorization header in the request with api key like this "ApiKey ababbababba"
5. Verify that request goes through and monitor is created

Testing behavior for users who do not have api key permission
---
1. Create a new role that has Uptime all permissions and synthetics-* index permissions
2. Create a new user with that role
3. Log in with that new user
4. Navigate to Uptime Monitor Management List. Click the api keys button.
5. Observe the different content for users without api key permissions.
